conanfile.py

```diff
@@ -0,0 +1,243 @@
+        self.requires("libcurl/[~7.64.1]")
+        self.options['libcurl'].shared = True
+        self.requires("zlib/[~1.2.8]")
+        self.options['zlib'].shared = True
+        self.requires("libuuid/1.0.3")
+        self.options['libuuid'].shared = True
+        self.requires("libzip/[~1.7.3]")
+        self.options['libzip'].shared = True
+        self.options['openssl'].shared = True
+        self.options['libxml2'].shared = True
+        if self.options.enable_testing:
+            self.options['gtest'].shared = True
+        if self.options.celix_add_openssl_dep:
+            self.requires("openssl/1.1.1k")
+        if self.options.build_remote_service_admin or self.options.build_shell_bonjour:
+            self.requires("libxml2/[~2.9.9]")
+        if self.options.build_cxx_remote_service_admin:
+            self.requires("rapidjson/[~1.1.0]")
+        if self.options.build_shell_bonjour:
+            # TODO: CC=cc is fixed in the official mdnsresponder Makefile, patching is needed to make cross-compile work
+            # https://github.com/conan-io/conan-center-index/issues/9711
+            # Another issue is in conan infrastructure: https://github.com/conan-io/conan-center-index/issues/9709
+            self.requires("mdnsresponder/1310.140.1")
+        if self.options.build_pubsub_psa_zmq:
+            self.requires("zeromq/4.3.2")
+            self.options['zeromq'].shared = True
+            self.requires("czmq/4.2.0")
+            self.options['czmq'].shared = True
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        for opt, val in self.options.values.items():
+            self._cmake.definitions[opt.upper()] = self.options.get_safe(opt, False)
+        self._cmake.definitions["CMAKE_PROJECT_Celix_INCLUDE"] = os.path.join(self.build_folder, "conan_paths.cmake")
+        # the following is workaround for https://github.com/conan-io/conan/issues/7192
+        # self._cmake.definitions["CMAKE_EXE_LINKER_FLAGS"] = "-Wl,--unresolved-symbols=ignore-in-shared-libs"
+        self.output.info(self._cmake.definitions)
+        v = tools.Version(self.version)
+        self._cmake.configure(defs={'CELIX_MAJOR': v.major, 'CELIX_MINOR': v.minor, 'CELIX_MICRO': v.patch})
+        return self._cmake
+
+    def build(self):
+        # self._patch_sources()
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("LICENSE", dst="licenses", src=self.source_folder)
+        cmake = self._configure_cmake()
+        cmake.install()
+
+    def package_info(self):
+        # enable [imports] in conanfile.txt and imports() of conanfile.py to collect
+        # not only executables and shared objects but also bundles
+        # this ability of get runtime artifacts out of Conan is crucial for integration with existing infrastructure
+        self.cpp_info.bindirs = ["bin", os.path.join("share", self.name, "bundles")]
```

Comment:
This line is worth looking at, and is subject to future improvement.

Reply:
Good to known. 

Celix also installs / uses a CelixTargets.cmake file so that it is possible to use bundles as imported CMake targets. 
I assume this also needs some attention if we want Celix to be available on conan central and as result we cannot depend on a CelixConfig.cmake file to include the CelixTargets.cmake file.